lib/auth/auth_with_roles.go

```diff
@@ -4468,8 +4471,11 @@ func (a *ServerWithRoles) ListAppSessions(ctx context.Context, pageSize int, pag
 
 // GetSnowflakeSessions gets all Snowflake web sessions.
 func (a *ServerWithRoles) GetSnowflakeSessions(ctx context.Context) ([]types.WebSession, error) {
-	if err := a.action(apidefaults.Namespace, types.KindDatabase, types.VerbList, types.VerbRead); err != nil {
-		return nil, trace.Wrap(err)
+	// Check if this a database service.
+	if !a.hasBuiltinRole(types.RoleDatabase) {
+		if err := a.action(apidefaults.Namespace, types.KindWebSession, types.VerbList, types.VerbRead); err != nil {
```

Comment:
Since this API should be called by DB service anyways, do we need to keep the `a.action()` code anymore? Should we at least add `REMOVE IN 14.0` at least?

Reply:
I prefer not to privilege a builtin role over normal RBAC - I think we've tried to move away from doing things like that. The builtin role check here is just a hack to allow db role access to snowflake web sessions... I can add a `REMOVE IN v14.0` comment I suppose, we can revisit it later.

I'd actually like to make the builtin db role have read/write access to web sessions with a conditional `where: equals(resource.sub_kind, "snowflake_session")`, i.e. only allow `KindWebSession` access when web sessions are for snowflake. I just don't want to widen the scope of this PR to do something like that. Although we really should be leveraging predicates rather than hardcoding special access to builtin roles imo. Maybe something for an RFD?